editor/src/clj/editor/atlas.clj

```diff
@@ -530,28 +594,18 @@
   (output texture-profile g/Any (g/fnk [texture-profiles resource]
                                   (tex-gen/match-texture-profile texture-profiles (resource/proj-path resource))))
 
-  (output all-atlas-images           [Image]             :cached (g/fnk [animations]
-                                                                   (into [] (comp (mapcat :images) (distinct)) animations)))
-
-  (output texture-set-data-generator g/Any (g/fnk [_node-id animations all-atlas-images extrude-borders inner-padding margin resource :as args]
-                                                  (or (when-let [errors (->> [[margin "Margin"]
-                                                                              [inner-padding "Inner Padding"]
-                                                                              [extrude-borders "Extrude Borders"]]
-                                                                             (keep (fn [[v name]]
-                                                                                     (validation/prop-error :fatal _node-id :layout-result validation/prop-negative? v name)))
-                                                                             not-empty)]
-                                                        (g/error-aggregate errors))
-                                                      {:f    generate-texture-set-data
-                                                       :args (assoc args
-                                                               :workspace (resource/workspace resource))})))
-
-  (output texture-set-data g/Any               :cached (g/fnk [texture-set-data-generator] (call-generator texture-set-data-generator)))
-  (output layout-size      g/Any               (g/fnk [texture-set-data] (:size texture-set-data)))
+  (output all-atlas-images [Image] :cached (g/fnk [animation-images]
+                                             (vec (distinct (flatten animation-images)))))
+
+  (output layout-data-generator g/Any          produce-layout-data-generator)
+  (output texture-set-data g/Any               :cached produce-texture-set-data)
+  (output layout-data      g/Any               :cached (g/fnk [layout-data-generator] (call-generator layout-data-generator)))
+  (output layout-size      g/Any               (g/fnk [layout-data] (:size layout-data)))
   (output texture-set      g/Any               (g/fnk [texture-set-data] (:texture-set texture-set-data)))
-  (output uv-transforms    g/Any               (g/fnk [texture-set-data] (:uv-transforms texture-set-data)))
-  (output layout-rects     g/Any               (g/fnk [texture-set-data] (:rects texture-set-data)))
+  (output uv-transforms    g/Any               (g/fnk [layout-data] (:uv-transforms layout-data)))
+  (output layout-rects     g/Any               (g/fnk [layout-data] (:rects layout-data)))
```

Comment:
We don't need the full `texture-set-data` for this, the `layout-data` (which is not dependent on the animation metadata) will suffice.